### PR TITLE
 refactor(storage-transfer): convert ava tests to mocha

### PR DESCRIPTION
**Pull request by nareshqlogic**
Tracking issue: Convert tests from ava to mocha [#1080](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/issues/1080) (it's a good idea to open an issue first for discussion)

### CODE REVIEW

**Comment by nareshqlogic** [comment]
Hi @fhinkel ,

I think 'Internal CI (storage-transfer)' — Tests failed due to insufficient permissions. Please verify that the necessary permissions have been granted.

  1) should create a storage transfer job:
     Error: Failed to obtain the location of the Google Cloud Storage (GCS) bucket nodejs-docs-samples-test-6f77307c-91c1-4059-8020-24b7171336d1 due to insufficient permissions.  Please verify that the necessary permissions have been granted.
      at Gaxios.<anonymous> (node_modules/gaxios/build/src/gaxios.js:72:27)
      at Generator.next (<anonymous>)
      at fulfilled (node_modules/gaxios/build/src/gaxios.js:16:58)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:189:7)

**Comment by fhinkel** [comment]
All tests pass without the mocha change, I think there's a problem with the refactoring. Let me take a look.

File: storage-transfer/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
   },
   "devDependencies": {
     "@google-cloud/nodejs-repo-tools": "^3.0.0",
-    "ava": "0.25.0",
+    "mocha": "^5.2.0",
```
Comment (fhinkel) [inline storage-transfer/package.json]: We had another PR updating `mocha` to `6.0`, please update here as well.

**Comment by fhinkel** [comment]
@nareshqlogic You changed the logic in `before()`, adding additional `await`s. I fixed it in https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/1202.

**Pull request closed without merging**